### ISSUE TITLE
Update EKAT to use Kokkos v3.5

### ISSUE THREAD

**Issue by jeff-cohere** [eagles-project/haero#328]
@mschmidt271 's work with TChem is starting to yield some interesting results. Recently, he has started using a version of TChem that requires the latest and greatest release of Kokkos (v3.5). In general, the TChem team tracks Kokkos much more closely than E3SM does, which places Haero in an occasionally awkward position, since Haero inherits the version of Kokkos that E3SM uses through the EKAT submodule.

**We are interested in updating EKAT's version of Kokkos to 3.5, but we can't do this unilaterally.** It's somewhat subtle to even do it in our own branch, because EKAT doesn't even track the official [kokkos/kokkos](https://github.com/kokkos/kokkos) repo--instead, it uses [E3SM's fork](https://github.com/E3SM-Project/kokkos).

For the moment, Mike is working with his own branch of EKAT in which he's switched back to the official Kokkos repo for the submodule. I think this is actually okay for now, but it doesn't allow us to merge Mike's changes unless we want to adopt his branch of EKAT. On the E3SM side of things, the SCREAM team (the other EKAT stakeholders) seem to be very busy fixing bugs in preparation for SCREAM's v1 release, so I hesitate to bug the relevant people with an issue like this that probably requires a bit of care.

When the time is right, here's how I envision us obtaining a version of Kokkos v3.5 in HAERO in a way that doesn't force us to switch over to the original Kokkos GitHub repo (which puts us at odds with SCREAM and E3SM):

1. We fetch the latest updates from the official repo to the E3SM fork of Kokkos. This makes the newer releases of Kokkos available to EKAT via its submodule, without changing the repo it's pointing at. We might need buy-in from the E3SM/SCREAM folks before doing this (because submodules are complicated and poorly abstracted, and this may cause issues, or at the very least confuse people).
2. We create a branch in EKAT (`kokkos-v3.5`, say) in which we fast-forward the `kokkos` submodule to point to the v3.5 release tag.
3. We push this new `kokkos-v3.5` EKAT branch and set Haero's EKAT submodule to it.

At this point, Haero can use Kokkos v3.5 while the SCREAM/E3SM people convince themselves that Kokkos v3.5 is ready for their own usage. In the past, they've had to make changes to Kokkos to get SCREAM to work with it, which is why there's an E3SM-specific fork of Kokkos in the first place.

TL;DR: I don't think we can act on this issue yet, but I believe we understand what needs doing. I'll try to gauge the interest of the SCREAM team on a quasi-weekly basis to make them aware of our intentions.
